nexus/src/app/background/sync_switch_port_settings.rs

```diff
@@ -0,0 +1,1236 @@
+                            continue;
+                        },
+                    };
+                    for config in configs {
+                        info!(
+                            &log,
+                            "applying bgp config";
+                            "switch_location" => ?location,
+                            "config" => ?config,
+                        );
+                        if let Err(e) = client.inner.bgp_apply(config).await {
+                            error!(log, "error while applying bgp configuration"; "error" => ?e);
+                        }
+                    }
+                }
+
+                //
+                // calculate and apply bootstore changes
+                //
+
+                // check downstream bootstore version
+                let mut downstream_config: Option<EarlyNetworkConfig> = None;
+
+                // Since we update the first scrimlet we can reach (we failover to the second one
+                // if updating the first one fails) we need to check them both.
+                for (_location, client) in &sled_agent_clients {
+                    let scrimlet_cfg  = match client.read_network_bootstore_config_cache().await {
+                        Ok(config) => config,
+                        Err(e) => {
+                            error!(log, "unable to read bootstore config from scrimlet"; "error" => ?e);
+                            continue;
+                        }
+                    };
+                    if let Some(other_config) = downstream_config.as_mut() {
+                        if other_config.generation < scrimlet_cfg.generation {
+                            *other_config = scrimlet_cfg.clone();
+                        }
+                    } else {
+                        downstream_config = Some(scrimlet_cfg.clone());
+                    }
+                }
+
+                // Move on to the next rack if neither scrimlet is reachable.
+                // if both scrimlets are unreachable we probably have bigger problems on this rack
+                if downstream_config.is_none() {
+                    error!(log, "both scrimlets are unreachable, cannot update bootstore");
+                    continue;
+                }
+
+                // get the upstream version
+
+                let subnet = match rack.rack_subnet {
+                    Some(IpNetwork::V6(subnet)) => subnet,
+                    Some(IpNetwork::V4(_)) => {
+                        error!(log, "rack subnet must be ipv6"; "rack" => ?rack);
+                        continue;
+                    },
+                    None => {
+                        error!(log, "rack subnet not set"; "rack" => ?rack);
+                        continue;
+                    }
+                };
+
+                // TODO: @rcgoodfellow is this correct? Do we place the BgpConfig for both switches in a single Vec to send to the bootstore?
```

Comment:
The primary concern is that this is consistent with that the sled-agent is expecting during early networking. So that is where I would go check.